design/ceph-config-updates.md

```diff
@@ -0,0 +1,235 @@
+  cannot connect to the wrong cluster accidentally
+- `fatal signal handlers = false`
+  **TODO MORE INFORMATION NEEDED** why does Rook set this? It's purpose will determine its config location
+
+#### Additional configs for test environments
+
+Test (especially integration tests) may need to specify `osd pool default size = 1` and
+`osd pool default min size = 1` to support running clusters with only one osd. Test environments
+would have a means of doing this fairly easily using the still-existent config override capability.
+Setting these values to this low value by default for production clusters may allow admins to create
+their own pools which are not fault tolerant accidentally.
+
+There is an option to set these values automatically for clusters which run with only one osd or to
+set this value for clusters with a number of osds less than the default programmatically within
+Rook's operator; however, this adds an additional amount of code flow complexity which is
+unnecessary except in tge integration test\ environments or in minimal demo environments. A middle
+proposed herein is to add `osd pool default {,min} size = 1` to the cluster CRD in Rook's
+`cluster/example/kubernetes/ceph` directory so that users "just trying out Rook" still get today's
+easy experience but where production clusters do not by default allow users to run clusters with
+potentially dangerous settings.
+
+### Minimal command line options
+
+The minimal set of command line options is pared down to the settings which inform the daemon where
+to find and/or store data.
+
+Required flags:
+- `--{public,cluster}-addr` are not required for most daemons but ...
+  - `--public-addr` and `--public-bind-addr` are necessary for mons
+  - `--public-addr` and `--cluster-addr` may be needed for osds
+    **TODO** find out for sure ^
+  - it may be more appropriate to set `--{public,cluster}-network` instead of `-addr`
+- `--keyring` may necessary to inform daemons where to find the keyring which must be mounted to a
+  nonstandard directory by virtue of being sourced via a Kubernetes secret.
+  - The keyring could copied from the secret mount location to Ceph's default location with an init
+    container, and for some daemons, this may be necessary. This should not be done for the mons,
+    however, as the mon keyrings also include the admin keyring, and persisting the admin key to
+    disk should be avoided at all costs for security.
+
+Notable non-required flags:
+- `--{mon,osd,mgr,mds,rgw}-data-dir` settings exist for all daemons, but it is more desirable to use
+  the `/var/lib/ceph/<daemon_type>/ceph-<daemon_id>` directory for daemon data within containers. If
+  possible, mapping the `dataDirHostPath/<rook_daemon_data_dir>` path on hosts to this default
+  location in the container is preferred.
+  - Note that currently, `dataDirHostPath` is mapped directly to containers, meaning that each
```

Comment:
Note that only the mon and osd pods currently require the dataDirHostPath. The other pods shouldn't need that host path mounted.